docs/auth/README.adoc

```diff
@@ -0,0 +1,89 @@
+= AeroGear Services Auth SDK
+
+Mobile authentication SDK based on link:http://www.keycloak.org/[Keycloak] using link:http://openid.net/connect/[OpenID Connect].
+
+== Usage
+
+To use the Auth SDK you'll first need to:
+
+* Have a Keycloak instance. See the <<Setting up Keycloak>> section.
+* Import the Core module. See link:./core/README.adoc[the Core documentation].
+* A configuration file added to the app directory. See the <<Configuration File>> section.
+
+=== Setting up Keycloak
+
+==== Using OpenShift
+
+* If you do not have mobile services enabled in your openshift cluster follow this link:https://github.com/aerogear/mobile-core/blob/master/docs/walkthroughs/local-setup.adoc[Local Setup] guide.
+* Navigate to your Openshift cluster and in the Service Catalog search for the Keycloak service.
+* Click on the Keycloak service and you will be prompted to fill in details about your app.  For now you can leave these as they are.  Navigate through the setup and click Create.
+This will provision the Keycloak service in the project you specified and create a public Client to be used with an app along with a bearer-only client.
+See link:http://www.keycloak.org/docs/latest/server_admin/index.html#oidc-clients[Keycloak OIDC client documentation].
+
+After provisioning, the Keycloak service will be available at the exposed Route. You can view this route inside your project or use the below command to get the route:
+----
+oc get route keycloak --template "http://{{.spec.host}} "
+----
+The route should look like `https://keycloak-myproject.192.168.37.1.nip.io/auth/`. +
+
+==== Standalone
+
+To setup standalone Keycloak follow Keycloak's guide link:/https://github.com/keycloak/keycloak/blob/master/README.md[here].
+
+=== Adding Auth Package
+Add the Auth package with NPM:
+----
+npm install aerogear/auth@latest --save
+----
+
+=== Configuration File
+
+A `mobile-services.json` file must exist in the app directory. It should specify configuration
+for Keycloak. This configuration can be generated by the link:https://github.com/aerogear/mobile-cli[AeroGear Mobile CLI].
+
+For an example of Keycloak configuration see the example apps link:../../example/cordova/www/mobile-services.json[mobile-services.json].
+
+The Auth SDK will use this configuration to communicate with Keycloak.
+
+
+=== Initializing the SDK
+`AuthService` can be retrieved by importing it:
+----
+import { AuthService } from 'aerogear/auth';
+----
+Before the `AuthService` can be used, the `init` function must be invoked once in an app.
+----
+let authService = new AuthService(keycloakConfig)
+authService.init()
```

Comment:
.init takes the onLoad config

Reply:
@TommyJ1994 I have been changed api slightly after integration with core parser. Let's merge that change (see separate PR) and we can update this one.